### PR TITLE
Append the legacy CRD using a Json patch

### LINKED ISSUE #9826: Generate CRDs compatible with the previous
### Description

The new operator CRDs are incompatible with the old one hence you need to remove the old CRDs before installing the new.
This is a problem and have possibly bad consequences when switching from one operator to the other.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by andreaTP**
Resolves: #9826 

Notes:
 - the CRD is copy pasted from [current `main`](https://github.com/keycloak/keycloak-operator/blob/main/deploy/crds/keycloak.org_keycloaks_crd.yaml)
 - extracted only the `/spec/versions/v1alpha1`
 - `served: false`
 - `storage: false`
 
We can automate the process of "downloading the old CRD and injecting it", but we need first some more clarity about how we want to release "kubernetes resources".

### CODE REVIEW

**Review by vmuzikar — CHANGES_REQUESTED** [review]
Thanks @andreaTP.

What is the expected result from this change? I tried the following:
1. `kubectl apply -f keycloak.org_keycloaks_crd.yaml` in the [old operator repo](https://github.com/keycloak/keycloak-operator/blob/main/deploy/crds/keycloak.org_keycloaks_crd.yaml).
2. `kubectl delete -f keycloak.org_keycloaks_crd.yaml`. This has no effect on the result.
3. `kubectl apply -k target` in the [new operator module](https://github.com/keycloak/keycloak/tree/main/operator).
4. `kubectl get keycloaks` resulting in this error:
```
Error from server (NotFound): Unable to list "keycloak.org/v1alpha1, Resource=keycloaks": the server could not find the requested resource (get keycloaks.keycloak.org)
```

File: operator/src/main/kubernetes/kustomization.yml [hunk 1]
```diff
@@ -9,3 +9,9 @@ resources:
 
 patches:
   - kubernetes/minikube.yml
+
+patchesJson6902:
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply operator/src/main/kubernetes/kustomization.yml]: Used `patches` but documentation is terribly scattered and added a comment to be able to use the old keywords.

**Comment by andreaTP** [comment]
@vmuzikar to test this change you miss to run a `mvn clean package` command, e.g:

1. kubectl apply -f keycloak.org_keycloaks_crd.yaml in the [old operator repo](https://github.com/keycloak/keycloak-operator/blob/main/deploy/crds/keycloak.org_keycloaks_crd.yaml).
2. mvn clean package in the [new operator module](https://github.com/keycloak/keycloak/tree/main/operator).
3. kubectl apply -k target in the [new operator module](https://github.com/keycloak/keycloak/tree/main/operator).

and the fact that it works shows that we fixed the issue 🙂 

Ready for a new round!

**Comment by andreaTP** [comment]
@vmuzikar any feedback on this one?

**Review by vmuzikar — APPROVED** [review]

**Review by abstractj — APPROVED** [review]

**Pull request merged**